segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/cache/BenchmarkTests.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.cache;
+
+import io.pravega.common.Exceptions;
+import io.pravega.common.Timer;
+import io.pravega.common.util.BufferView;
+import io.pravega.common.util.ByteArraySegment;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import lombok.Cleanup;
+import lombok.Data;
+import lombok.val;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Benchmark tests for {@link CacheStorage} and its implementations. This should be used to compare the runtime performance
+ * of various changes in the implementation. Since the results may vary significantly based on the hardware used, outcomes
+ * are not comparable across environments. The way to use this is to establish a benchmark using the base code and then
+ * re-run it with the modifications already implemented and tested.
+ *
+ * The number of iterations {@link #ITERATION_COUNT} should always be greater than 1 since the first iteration will incur
+ * the costs of the initial memory allocation, while the remaining ones will benefit from the memory already being allocated.
+ *
+ * This is marked as @Ignore since these are not real unit tests (no correctness checking) and they take a long time to execute.
+ */
+@Ignore
```

Comment:
Given that we are ignoring it when running junit, I'm wondering whether this is the right place for this benchmark. Is there are reason you want this file here? Could it be in `main` or even in another repository altogether like pravega-benchmark?

One concern I have with it being embedded deep down into the code repository is that only you will remember that this is here, while we might want to give some more visibility to it.

Reply:
It will have even less visibility if it's in some obfuscated `main` class. I like it as unit test because it's easy to run it from the IDE and execute the IDE's code coverage on it. 

This is a pattern I've used before (for Segment Attributes and for the `AVLTreeIndex`).

Reply:
Location matters if you want others to use it too. To clarify, when I said `main`, I was not referring to a `main` class, but instead the `src/java/main` directory instead of `src/java/test` so that it is not confused with ignored test code.

Feel free to resolve this thread once you've read this, I only wanted to clarify the point.

Reply:
This is good info. Thanks!